modules/savedb.py

```diff
@@ -0,0 +1,60 @@
+import os
+import socket
+import pymongo
+
+original_socket = socket.socket
+
+url = os.getenv("DATABASE_URL")
+passwd = os.getenv("DATABASE_PASSWORD")
+user = os.getenv("DATABASE_USERNAME")
+database = os.getenv("DATABASE_NAME")
+
+def check_db_options():
+    """ Just to check whether database
+    parameters are loaded
+    """
+    if not (url and passwd and user and database):
+        print("Could not load Database Configurations")
+        return 1
+
+def saveToDatabase(links):
+    """
+    Connects to a MongoDB
+    Create Mongo Collection
+    Add the links to the collection
+    Args:
+        url = url of the mongo server
+        database = data that is being stored in the database
+        user = username to login into Mongo
+        passwd = password of Mongo
+        link = URLs from the crawler
+    if not url and not database:
+        print("URL and DATABASE are null")
+        print("Links are not stored into database")
+        exit()
+    """
+    socket.socket = original_socket
+    if not user and not passwd:
+        client = pymongo.MongoClient(url)
+    elif url and database and user and passwd:
+        client = pymongo.MongoClient(url,
+                                     username=user,
+                                     password=passwd,
+                                     authSource=database,
+                                     authMechanism='SCRAM-SHA-256')
```

Comment:
Can we allow for different authentication mechanisms?

Reply:
yea there are other auth mechanisms we can use
http://api.mongodb.com/python/current/examples/authentication.html

Reply:
So you should use an optional parameter that allows the function to dynamically change `authMechanism`  to allow flexibility and you can use the default value of `'SCRAM-SHA-256'`

Reply:
If you'd like to add some robustness, you can add a check before using the `authMechanism` string to make sure that it's one of the approved mechanisms.

Reply:
If you'd like to add some robustness and add a good test case then you can add a check for the `authMechanism` parameter to make sure it's one of the valid authentication mechanisms being passed. It would be easy to write a test for once you figure out how to mock a DB connection.